lecture-notes/2-vcs.md

```diff
@@ -1673,10 +1673,94 @@ TODO: Talk about interactive rebasing for squashing/prettifying.
 
 CONTENTS: Collaboration with Git
 
+You've done enough solo development. It's now time to start working with your
+groupmates on your project. In order to do that, you need to make your
+repository accessible to the others somehow.
+
+Git is pretty flexible about how repositories can be shared---after all, they
+are "just files"---but the easiest and most common way to share is via a code
+forge such as GitHub, SourceHut, or GitLab. These Git forges do a bunch of
+stuff but the most important part is hosting a copy of your `.git` folder and
+managing permissions. It's totally possible to DIY this with your own server
+and software such as Gitolite (I used to do that for years), but these forges
+*also* provide issue tracking and pull requests and other niceties.
+
+So you go ahead and make a repository on GitHub. Now you're in this weird
+situation where you have a repository locally and GitHub has this blank
+repository somewhere in San Francisco and you have to find a way to get your
+code across the country. To do this, you need to add a *remote*.
+
 ### Sharing branches with others
 
-TODO: What's a remote? Example of `git remote add` and `git remote -v`. Talk
-about different remote protocols.
+A *remote* is a copy of your repository that is... somewhere else. It could be
+the directory next door, it could be a big binary file called a *bundle*,
+or---most commonly---it could be on someone else's computer[^xkcd908]
+accessible over the network. These network remotes look like an HTTPS remote
+`https://server.com/myusername/myreponame` or a SSH remote
+`git@server.com:myusername/myreponame.git` or (less commonly) a Git remote
+`git://server.com/myusername/myreponame.git`. These days, `server.com` is
+`github.com` or one of the other forges. These URLs specify the location and
+protocol for how your local Git installation should make network requests to
+the remote repository.
+
+[^xkcd908]: See [XKCD 908](https://xkcd.com/908/).
+
+One repository can have zero or more remotes, though having one remote is
```

Comment:
```suggestion
A repository can have zero or more remotes, though having one remote is
```

Reply:
maybe a local copy of the repo, since this config shit is local only

Reply:
Fair point, but I still think it's fine to say "a repository"

Reply:
I said "a repository on your local machine"